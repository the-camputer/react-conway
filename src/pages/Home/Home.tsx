import React from 'react';
import { Link } from 'react-router-dom';
import { Stack, Button, Typography } from '@mui/joy';
import { Info, PlayArrow } from '@mui/icons-material';

const Home: React.FC = (props: any) => {
  return (
    <div className='background'>
      <Stack height='100vh' justifyContent='space-around'>
        <Typography
          level='h1'
          sx={{
            color: 'white',
            WebkitTextStroke: '1px black',
            fontSize: '12em',
          }}
        >
          Game of Life
        </Typography>
        <Stack spacing={2}>
          <Link to={'/game'} data-testid='play-link'>
            <Button size='lg' startDecorator={<PlayArrow />}>
              Play
            </Button>
          </Link>
          <Link to={'/about'} data-testid='about-link'>
            <Button size='lg' startDecorator={<Info />}>
              About
            </Button>
          </Link>
        </Stack>
      </Stack>
    </div>
  );
};

export default Home;
