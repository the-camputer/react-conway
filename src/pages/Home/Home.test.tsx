import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Home from './Home';

describe('Home', () => {
  it('renders without crashing', () => {
    const view = render(<Home />, { wrapper: BrowserRouter });
    expect(view).toMatchSnapshot();
  });
});
