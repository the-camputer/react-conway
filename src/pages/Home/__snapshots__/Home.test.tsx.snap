// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders without crashing 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="background"
      >
        <div
          class="MuiStack-root css-1iatv4s-JoyStack-root"
        >
          <h1
            class="MuiTypography-root MuiTypography-h1 css-oc1cc4-JoyTypography-root"
          >
            Game of Life
          </h1>
          <div
            class="MuiStack-root css-hrqung-JoyStack-root"
          >
            <a
              data-testid="play-link"
              href="/game"
            >
              <button
                class="MuiButton-root MuiButton-variantSolid MuiButton-colorPrimary MuiButton-sizeLg css-1mhx1qt-JoyButton-root"
                type="button"
              >
                <span
                  class="MuiButton-startDecorator css-187ulk2-JoyButton-startDecorator"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="PlayArrowIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8 5v14l11-7z"
                    />
                  </svg>
                </span>
                Play
              </button>
            </a>
            <a
              data-testid="about-link"
              href="/about"
            >
              <button
                class="MuiButton-root MuiButton-variantSolid MuiButton-colorPrimary MuiButton-sizeLg css-1mhx1qt-JoyButton-root"
                type="button"
              >
                <span
                  class="MuiButton-startDecorator css-187ulk2-JoyButton-startDecorator"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="InfoIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
                    />
                  </svg>
                </span>
                About
              </button>
            </a>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="background"
    >
      <div
        class="MuiStack-root css-1iatv4s-JoyStack-root"
      >
        <h1
          class="MuiTypography-root MuiTypography-h1 css-oc1cc4-JoyTypography-root"
        >
          Game of Life
        </h1>
        <div
          class="MuiStack-root css-hrqung-JoyStack-root"
        >
          <a
            data-testid="play-link"
            href="/game"
          >
            <button
              class="MuiButton-root MuiButton-variantSolid MuiButton-colorPrimary MuiButton-sizeLg css-1mhx1qt-JoyButton-root"
              type="button"
            >
              <span
                class="MuiButton-startDecorator css-187ulk2-JoyButton-startDecorator"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="PlayArrowIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8 5v14l11-7z"
                  />
                </svg>
              </span>
              Play
            </button>
          </a>
          <a
            data-testid="about-link"
            href="/about"
          >
            <button
              class="MuiButton-root MuiButton-variantSolid MuiButton-colorPrimary MuiButton-sizeLg css-1mhx1qt-JoyButton-root"
              type="button"
            >
              <span
                class="MuiButton-startDecorator css-187ulk2-JoyButton-startDecorator"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="InfoIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
                  />
                </svg>
              </span>
              About
            </button>
          </a>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
